test/math.jl

```diff
@@ -760,6 +760,102 @@ end
     @test quadgk(cos, 0,0.7,1, norm=abs)[1] ≈ sin(1)
 end
 
+module Test19626
+    using Base.Test
+    # Begin tests for 19626: Unitful compatibility.
+    # Define a mock physical quantity type
+    immutable MockQuantity{T} <: Number
+        val::T
+    end
+    Base.promote_rule{T,S}(::Type{MockQuantity{T}}, ::Type{MockQuantity{S}}) =
+        MockQuantity{promote_type(T,S)}
+    Base.convert{T}(::Type{MockQuantity{T}}, x::MockQuantity) = MockQuantity(T(x.val))
+
+    # Following definitions needed for quadgk to work with MockQuantity
+    import Base: +, -, *, abs, isnan, isinf, isless, <=
+    import Base.QuadGK: quadgk, do_quadgk
+    +(a::MockQuantity) = a
+    +(a::MockQuantity, b::MockQuantity) = MockQuantity(a.val+b.val)
+    -(a::MockQuantity) = MockQuantity(-a.val)
+    -(a::MockQuantity, b::MockQuantity) = MockQuantity(a.val-b.val)
+    *(a::MockQuantity, b::Number) = MockQuantity(a.val*b)
+    *(b::Number, a::MockQuantity) = MockQuantity(b*a.val)
+    abs(a::MockQuantity) = MockQuantity(abs(a.val))
+    isnan(a::MockQuantity) = isnan(a.val)
+    isinf(a::MockQuantity) = isinf(a.val)
+    isless(a::MockQuantity, b::MockQuantity) = isless(a.val, b.val)
+
+    # isless defn. necessary so that default abstol plays nicely with MockQuantity
+    isless(y::Number, x::MockQuantity) = y == 0 ? isless(MockQuantity(0), x) :
+        error("Dimensions issue")
+    isless(x::MockQuantity, y::Number) = y == 0 ? isless(x, MockQuantity(0)) :
+        error("Dimensions issue")
+
+    function quadgk{T}(f, a::MockQuantity{T}, b::MockQuantity{T},
+        c::MockQuantity{T}...; abstol=zero(T), reltol=sqrt(eps(T)),
+        maxevals=10^7, order=7, norm=vecnorm)
+
+        _do_quadgk(f, [a, b, c...], order, T, abstol, reltol, maxevals, norm)
+    end
+
+    # Necessary with infinite or semi-infinite intervals since !(MockQuantity <: Real)
+    # and do_quadgk tests if eltype(s) <: Real.
+    function _do_quadgk{Tw,T<:Real}(f, s::Array{MockQuantity{T},1}, n, ::Type{Tw},
```

Comment:
This seems overly complicated.   For testing this PR, it should be sufficient to have an `f(x)` where the *integrand* `f` is a `MockQuantity`, but `x` is still `Float64`.